### DISCUSSION TITLE
Is there a way to put mock response in a file ?

### DISCUSSION THREAD

**Discussion by gillescrieloue** [Decathlon/tzatziki#327] [Q&A]
Hello everyone,

I currently use Wiremock and have all my mocks handled with files (in the mapping folder).
It's very useful, especially because my mocked partners responses can be very long.

With Tzatziki, Is there a way to not having the answers directly in cucumber (which is really not very readable in my case), but in separated files ?
I didn't find a way, I tried reading files and putting the result in variables : 

```
 Scenario:
    Given result1 is "{{{[&__files/result.json]}}}"
    And that getting on "endpoint" will return a status OK_200 and:
    """
    {{result1}}
    """
```
 
but variables don't seem to be available either in the mock answers.

Thanks

**Comment by alexandrepa** [comment]
Hi, normally it should work, are you sure Tzatziki is finding your json file ? Do you have an error on the first step ? Where is your result.json file located in your project ? 

The current path is located at the resources folder of your test folder if i'm right.

**Comment by gillescrieloue** [comment]
Hello, yes I'm sure the file is present and the variable in itself works :
I use the same strategy to compare the expected result of my endpoint to a result stored in a file and in this context it works.

But I never succeeded in using variables in mocks results. When debugging the tzatziki steps, I didn't see a variable replacement either.

**Comment by alexandrepa** [comment]
I did a test to mock the Scenario you post in your first message and it worked with a file. Can you share your code ?

**Reply by gillescrieloue** [reply]
I got back working on my branch this morning, and indeed, it's working correctly without making any changes.
I could not find why it wasn't working last time.

Thanks for your help and for confirming it's possible.
